server/src/main/java/org/opensearch/common/compress/CompressedXContent.java

```diff
@@ -169,6 +170,10 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeByteArray(bytes);
     }
 
+    public void writeVerifiableTo(BufferedChecksumStreamOutput out) throws IOException {
+        out.writeInt(crc32);
```

Comment:
I see that we are ignoring `bytes` field since that can be different. In that case, should we ingore crc32 as well since this is the checksum and this will be different as well if the bytes are different.